DateTimeCalendar/Calendar.java

```diff
@@ -0,0 +1,137 @@
+public class Calendar{
+    /*
+    Goals:
+    - When created, indicate date and current day of the week (Monday, Tuesday, Wednesday, etc…)
+    - Given another date tell which day of the week it was or it will be.
+    - Calculate number of weeks within a month of a certain year. A week counts if all seven 
+    days (monday to sunday) happen within the same month; if the month starts in the middle of 
+    the week that week is not considered for the number returned by this method.
+    */ 
+
+    public enum DayOfWeek {
+        SUN("Sunday"),
+        MON("Monday"),
+        TUE("Tuesday"),
+        WED("Wednesday"),
+        THR("Thursday"),
+        FRI("Friday"),
+        SAT("Saturday"),
+        ;
+
+        public final String longName;
+
+        private DayOfWeek(final String longName){
+            this.longName = longName;
+        }
+    }
+
+    private int currentDayOfWeek;
+    private Date date;
+
+    private Calendar(){
+
+    }
+
+    public Calendar(Date date){
+        setDate(date);
+        setCurrentDayOfTheWeek(date);
+    }
+
+    public static int calcWeeksInMonth(Date date){
+        //a week is considered mon-sun
+        
+        //Example: 3/1/1900
+        int monthDayCount = date.getMonth().days; // 31
+
+        //check for month feb and for leap year
+        if (date.getMonth().ordinal() == 1 && Date.isLeapYear(date.getYear())) {
+            monthDayCount++;
+        }
+
+        Date firstOfTheMonth = new Date(date.getYear(), date.getMonth(), 1); // creating obj to represent first of the month
+        int firstWeekDayOfMonth = calcDayOfWeek(firstOfTheMonth); // calculate first week day of the month -> 4 (Thursday)
+
+        //want to use first day of the month as a place holder while we tick ahead to tell us the day of the first monday
+        int firstMondayOfMonth = 1; //Every month starts on day 1
```

Comment:
This can be a constant in the class

Reply:
Why would it be preferred to be a constant over the current implementation?

Reply:
Going by just by reading the code and the comment: `int firstMondayOfMonth = 1; //Every month starts on day 1`.

So if _**every month will start on day 1**_, that sounds like a constant to me. But I see also that originally you were using this to do some calculations over the loop. After you refactor that piece of code to remove the loop we can revisit this comment :)